packages/@react-stately/collections/src/getChildNodes.ts

```diff
@@ -49,3 +50,36 @@ export function getLastItem<T>(iterable: Iterable<T>): T | undefined {
 
   return lastItem;
 }
+
+export function compareNodeOrder<T>(collection: Collection<Node<T>>, a: Node<T>, b: Node<T>) {
+  // If the two nodes have the same parent, compare their indices.
+  if (a.parentKey === b.parentKey) {
+    return a.index - b.index;
+  }
+
+  // Otherwise, collect all of the ancestors from each node, and find the first one that doesn't match starting from the root.
+  let aAncestors = getAncestors(collection, a);
+  let bAncestors = getAncestors(collection, b);
+  let firstNonMatchingAncestor = aAncestors.slice(0, bAncestors.length).findIndex((a, i) => a !== bAncestors[i]);
+
+  if (firstNonMatchingAncestor !== -1) {
+    // Compare the indices of two children within the common ancestor.
+    a = aAncestors[firstNonMatchingAncestor];
+    b = bAncestors[firstNonMatchingAncestor];
+    return a.index - b.index;
+  }
+
+  // 🤷
+  return -1;
```

Comment:
ah yeah true. I guess it would ideally compare the index of the first item in b's ancestors with the index of a itself in that case. But yeah we don't really support that case at the moment (and I'd like to disallow it actually).